src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -322,49 +322,45 @@ Klass* SystemDictionary::resolve_array_class_or_null(Symbol* class_name,
   return k;
 }
 
+static inline void log_circularity_error(PlaceholderEntry* probe) {
+  LogTarget(Debug, class, load, placeholders) lt;
+  if (lt.is_enabled()) {
+    ResourceMark rm;
+    LogStream ls(lt);
```

Comment:
Is it helpful to pass THREAD in for the ResourceMark?

Reply:
Yes, I have a THREAD handy.